marklogic-client-api/src/main/java/com/marklogic/client/datamovement/Splitter.java

```diff
@@ -1,13 +1,42 @@
+/*
+ * Copyright 2019 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.marklogic.client.datamovement;
 
 import java.io.InputStream;
 import java.util.stream.Stream;
 
 import com.marklogic.client.io.marker.AbstractWriteHandle;
 
+/**
+ * To facilitate CSV splitting, Splitter allows to implement the split method where the 
```

Comment:
Please describe Splitter as the abstract definition of a class that can split an input stream into a Java stream of write handles.  Typically, the handles are used for writing documents to the database.

(In other words, the abstraction shouldn't mention the concrete implementation -- that's why it's an abstraction.)